pkg/datapath/linux/ipsec/ipsec_linux.go

```diff
@@ -1121,3 +1121,20 @@ func StartStaleKeysReclaimer(ctx context.Context) {
 		}
 	}()
 }
+
+// We need to install xfrm state for the local router (cilium_host) early
+// in daemon init path. This is to ensure that we have the xfrm state in
+// place before we advertise the routerIP where other nodes may potentially
+// pick it up and start sending traffic to us. This was previously racing
+// and creating XfrmInNoState errors because other nodes picked up node
+// update before Xfrm config logic was in place. So special case init the
+// rule we need early in init flow.
+func Init() error {
+	outerLocalIP := node.GetInternalIPv4Router()
+	wildcardIP := net.ParseIP("0.0.0.0")
+	localCIDR := node.GetIPv4AllocRange().IPNet
+	localWildcardIP := &net.IPNet{IP: wildcardIP, Mask: net.IPv4Mask(0, 0, 0, 0)}
+
+	_, err := UpsertIPsecEndpoint(localCIDR, localWildcardIP, outerLocalIP, widcardIP, 0, IPSecDirIn, false)
```

Comment:
Should we consider the situation of subnetEncryption?

I ask this because we have different code logic at `linuxNodeHandler.enableIPsecIPv4()`: 

1\. `if n.subnetEncryption()`, we use `nodeConfig.IPv4PodSubnets` as remoteCIDR
https://github.com/cilium/cilium/blob/6fcad1351103594602767024c1e2d17816f69f34/pkg/datapath/linux/node.go#L1113-L1119

2\. `else`, we use localCIDR:
https://github.com/cilium/cilium/blob/6fcad1351103594602767024c1e2d17816f69f34/pkg/datapath/linux/node.go#L1133

Reply:
Yes we should also fix subnetEncryption case the race is much tighter though because it would require some other node to learn the CIDRs before we set them from local state. Its still out of order init though IMO and should be done from this init() call. I'll argue a follow up fix should handle though. Happy to do it next.

Reply:
Filed #28019 to track the follow-up fix. Resolving the conversation now.